packages/next-app/src/components/Sidebar.tsx

```diff
@@ -0,0 +1,131 @@
+"use client";
+
+import { MagnifyingGlassIcon } from "@radix-ui/react-icons";
+import classNames from "classnames";
+import { createContext, useContext } from "react";
+
+interface SidebarContextType {
+  isFloating: boolean;
+}
+
+const SidebarContext = createContext({} as SidebarContextType);
+
+function useSidebarContext() {
+  const context = useContext(SidebarContext);
+
+  if (!context) {
+    throw new Error(
+      "Child components of Sidebar cannot be rendered outside the Sidebar component!"
+    );
+  }
+
+  return context;
+}
+
+export default function Sidebar({
+  children,
+  isFloating = false,
+}: {
+  children: React.ReactNode;
+  isFloating?: boolean;
+}) {
+  return (
+    <SidebarContext.Provider value={{ isFloating }}>
+      <div
+        className={classNames(
+          "w-full max-w-full  py-5",
+          isFloating ? "bg-gray-900" : "bg-gray-800"
+        )}
+      >
+        <div className="h-screen w-96 max-w-full items-start justify-start space-y-4">
+          {children}
+        </div>
+      </div>
+    </SidebarContext.Provider>
+  );
+}
+
+function Header({
+  children,
+  name,
+}: {
+  children?: React.ReactNode;
+  name: string;
+}) {
+  useSidebarContext();
+  return (
+    <div className="items-start justify-start space-y-2.5 self-stretch px-5">
+      <div className="flex items-center justify-start space-x-0 text-2xl font-medium text-gray-400">
+        <span>{name}</span>
+      </div>
+      {children}
+    </div>
+  );
+}
+
+function Content({ children }: { children: React.ReactNode }) {
+  const { isFloating } = useSidebarContext();
+  return (
+    <div
+      className={classNames(
+        "relative max-h-[40rem] self-stretch overflow-auto",
+        isFloating ? "rounded-md border border-gray-700 bg-gray-800 mx-5" : ""
+      )}
+    >
+      {children}
+    </div>
+  );
+}
+
+function InputFilter({
+  placeHolder,
+  onChange,
+}: {
+  placeHolder: string;
+  onChange: (e: React.FormEvent<HTMLInputElement>) => void;
+}) {
+  useSidebarContext();
```

Comment:
👀 is this supposed to be here?

Reply:
Do you mean the `useSidebarContext();` or the `InputFilter`?  If we see `InputFilter` being applicable somewhere else then I would transfer this as a standalone component

Reply:
I mean what's the purpose of calling this hook here with no return? 🥸

Reply:
aaa this hook is to make sure that sidebar sub-components cannot be used outside `<Sidebar>`

without it, you would be able to use these components everywhere without `<Sidebar>`
Sidebar.Header = Header;
Sidebar.Content = Content;
Sidebar.InputFilter = InputFilter;
Sidebar.Item = Item;

Reply:
And would that really be an issue in your opinion? 😅